### ISSUE TITLE
what is the function of the "final_crop"

### ISSUE THREAD

**Issue by yitang-hnu** [saic-vul/iterdet#15]
Hello, 
Thanks for your research and for open-sourcing the code. I found there are some extra kwargs like "final_crop" in your config, and what is the function of the "final_crop" in rpn_head,  bbox_head and resize pipeline separately？

**Comment by filaPro** [comment]
Hi, @yitang-hnu 

Looks like by default some classes of `mmdetection` finally crop all boxes by the image bounds. However, CrowdHuman dataset with _full_ body annotations contains a couple of partially out-of-bound boxes. So with this `final_crop` flag we enable the detector to predict these boxes and achieve slightly better scores.

**Comment by yitang-hnu** [comment]
Got it. Thx!

**Comment by hh23333** [comment]
Hi, thanks for sharing your excellent work. I trained the faster-rcnn baseline on crowdhuman dataset. I followed the same setting with iterdet. And got such results: recall: 0.8713, mAP: 0.8383, mMR: 0.4946. It seems that the mmr is normal but the recall and mAP are far from satisfactory, can you give me some suggestions? Thanks a lot.

**Comment by filaPro** [comment]
Hi @hh23333 ,
Have you just run: `bash tools/dist_train.sh configs/iterdet/crowd_human_full_faster_rcnn_r50_fpn_2x.py 8 --validate` on 8 GPUs? It's strange as we did the same and provide the model with higher scores.

**Comment by hh23333** [comment]
Hi, @filaPro.
Thanks for your prompt reply. I replaced the 'IterDetFasterRCNN' with normal faster-rcnn, and re-employ it with a newer version of mmdetection. I also replaced the 'final crop' option in rpn_head and bbox_head with 'clip_border' in bbox_coder and 'final_crop' with 'bbox_clip_border' in Resize transform, as they have the same function. Other parameters are the same as it in 'crowd_human_full_faster_rcnn_r50_fpn_2x.py'.

**Comment by filaPro** [comment]
May be it is better to follow original `FasterRCNN` config, not `IterDetFasterRCNN` for such experiments. For example all configs in `mmdetection` implies `SGD` optimizer with `lr warmup` as we are using `Adam` with different learning rate, etc.
